packages/shorebird_cli/test/src/commands/patch/patch_ios_framework_command_test.dart

```diff
@@ -0,0 +1,572 @@
+https://github.com/shorebirdtech/shorebird/issues/472
+'''),
+      ).called(1);
+    });
+
+    test('throws error when creating aot snapshot fails', () async {
+      const error = 'oops something went wrong';
+      when(() => aotBuildProcessResult.exitCode).thenReturn(1);
+      when(() => aotBuildProcessResult.stderr).thenReturn(error);
+      final tempDir = setUpTempDir();
+      setUpTempArtifacts(tempDir);
+      final exitCode = await IOOverrides.runZoned(
+        () => runWithOverrides(command.run),
+        getCurrentDirectory: () => tempDir,
+      );
+      verify(
+        () => progress.fail('Exception: Failed to create snapshot: $error'),
+      ).called(1);
+      expect(exitCode, ExitCode.software.code);
+    });
+
+    test('does not create patch on --dry-run', () async {
+      when(() => argResults['dry-run']).thenReturn(true);
+      final tempDir = setUpTempDir();
+      setUpTempArtifacts(tempDir);
+      final exitCode = await IOOverrides.runZoned(
+        () => runWithOverrides(command.run),
+        getCurrentDirectory: () => tempDir,
+      );
+      expect(exitCode, equals(ExitCode.success.code));
+      verifyNever(
+        () => codePushClientWrapper.createPatch(
+          appId: any(named: 'appId'),
+          releaseId: any(named: 'releaseId'),
+        ),
+      );
+      verify(() => logger.info('No issues detected.')).called(1);
+    });
+
+    test('does not prompt on --force', () async {
+      when(() => argResults['force']).thenReturn(true);
+      final tempDir = setUpTempDir();
+      setUpTempArtifacts(tempDir);
+      final exitCode = await IOOverrides.runZoned(
+        () => runWithOverrides(command.run),
+        getCurrentDirectory: () => tempDir,
+      );
+      expect(exitCode, equals(ExitCode.success.code));
+      verifyNever(() => logger.confirm(any()));
+      verify(
+        () => codePushClientWrapper.publishPatch(
+          appId: any(named: 'appId'),
+          releaseId: any(named: 'releaseId'),
+          platform: any(named: 'platform'),
+          channelName: any(named: 'channelName'),
+          patchArtifactBundles: any(named: 'patchArtifactBundles'),
+        ),
+      ).called(1);
+    });
+
+    test('succeeds when patch is successful', () async {
+      final tempDir = setUpTempDir();
+      setUpTempArtifacts(tempDir);
+      final exitCode = await IOOverrides.runZoned(
+        () => runWithOverrides(command.run),
+        getCurrentDirectory: () => tempDir,
+      );
+      verify(
+        () => logger.info(
+          any(
+            that: contains(
+              '''🕹️  Platform: ${lightCyan.wrap('ios')} ${lightCyan.wrap('[aarch64 (0 B)]')}''',
+            ),
+          ),
+        ),
+      ).called(1);
+      verify(() => logger.success('\n✅ Published Patch!')).called(1);
+      expect(exitCode, ExitCode.success.code);
+    });
+
+    test(
+        'succeeds when patch is successful '
+        'with flavors and target', () async {
```

Comment:
can we either support flavor/target as options or remove this?

Reply:
Done.